### PR TITLE
upgrade to sc4, fix bundle, more

### PR DESCRIPTION
**Pull request by brad-decker**
1. Upgrade styled-components to 4 @ beta
2. Refactor extend out
3. Refactor injectGlobal in favor of exporting OPTIONAL global style components
4. Add Rollup that creates two single file bundles
5. Make /pages use our built files instead of src to identify bundle issues sooner.

To do:
- [x] refactor tooltip to use refs correctly
- [x] update docs
- [x] update scripts
- [x] determine best development workflow using rollup + next
- [x] Test in production.

### CODE REVIEW

**Comment by brad-decker** [comment]
1.5K of the diff is in the yarn.lock. most of the files come from changing three lines in all of the icons :(

**Comment by brad-decker** [comment]
@gregorelee and @danieltisdale done. I made significant changes to the behind-the-scenes stuff with the menu. @gregorelee tag me when you get a minute today and i'll walk you through the expected behavior. @AriLFrankel @ceciliamvrie @martahj might want to tag you on this too make sure i didn't do anything stupid ;)

**Comment by ceciliamvrie** [comment]
@brad-decker I'm encountering a bug with toggling back the persistent bottom sheet. It works fine when I open it, but it covers 2/3rds of the browser when I try to close it.

**Comment by brad-decker** [comment]
@ceciliamvrie thanks! Resolved this one too

File: src/enhancers/withRipple.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ const rippleAnimation = keyframes`
 `;
 
 const withRipple = (ComposedComponent) => {
-  const RippledComponent = ComposedComponent.extend`
+  const RippledComponent = styled(ComposedComponent)`
```
Comment (petegivens) [inline src/enhancers/withRipple.js]: Were these changes performed by the SC automatic codemod? Looks like we're missing about 900 icons =x

File: pages/_app.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    return { pageProps };
+  }
+
+  render() {
+    const { Component, pageProps } = this.props;
+    return (
+      <Container>
+        <SMCGlobalStyles />
```
Comment (petegivens) [inline pages/_app.js]: can these be edited anywhere in the tree by importing `SMCGlobalStyles` and overwriting values in the css block?

File: src/enhancers/withRipple.js (hunk 1 above)
Comment (brad-decker) [reply src/enhancers/withRipple.js]: what do you mean by missing? I did this using regex.

File: pages/_app.js (hunk 2 above)
Comment (brad-decker) [reply pages/_app.js]: these are full on styled components, can take props, i'm not sure about using styled() or createGlobalStyles to extend them

File: src/enhancers/withRipple.js (hunk 1 above)
Comment (petegivens) [reply src/enhancers/withRipple.js]: diff is showing that only 43 icon files were changed. shouldn't it be all of them? 
image

Comment (brad-decker) [reply src/enhancers/withRipple.js]: if you're using that tree addon it can only load 100 files ;)

Comment (petegivens) [reply src/enhancers/withRipple.js]: nothing to see here, folks.

**Pull request merged**